test/src/test/java/org/axonframework/test/aggregate/ResultValidatorImplTest.java

```diff
@@ -72,9 +96,150 @@ void shouldConsiderExplicitEqualsBeforeCheckingFields() {
         validator.expectEvents(s2);
     }
 
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlineInsideWindow() {
+        Instant expiryTime = deadlineWindowFrom.plus(1, ChronoUnit.DAYS);
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(createDeadline(expiryTime)));
+
+        assertThrows(AxonAssertionError.class, () -> validator.expectNoScheduledDeadlineMatching(deadlineWindowFrom, deadlineWindowTo, Matchers.anything()));
+    }
+
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlineAtFrom() {
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(createDeadline(deadlineWindowFrom)));
+
+        assertThrows(AxonAssertionError.class, () -> validator.expectNoScheduledDeadlineMatching(deadlineWindowFrom, deadlineWindowTo, Matchers.anything()));
+    }
+
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlineAtTo() {
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Collections.singletonList(createDeadline(deadlineWindowTo)));
+
+        assertThrows(AxonAssertionError.class, () -> validator.expectNoScheduledDeadlineMatching(deadlineWindowFrom, deadlineWindowTo, Matchers.anything()));
+    }
+
+    @Test
+    void noDeadlineMatchingInTimeframeWithDeadlinesOutsideWindow() {
+        ScheduledDeadlineInfo deadlineBefore = createDeadline(deadlineWindowFrom.minus(1, ChronoUnit.DAYS));
+        ScheduledDeadlineInfo deadlineAfter = createDeadline(deadlineWindowTo.plus(1, ChronoUnit.DAYS));
+        when(deadlineManager.getScheduledDeadlines()).thenReturn(Arrays.asList(deadlineBefore, deadlineAfter));
+
+        validator.expectNoScheduledDeadlineMatching(deadlineWindowFrom, deadlineWindowTo, Matchers.anything());
```

Comment:
Just a thought, but how about we do an `assertDoesNotThrow`?
I think this only makes sense because all the other methods do assert an exception to be thrown.
And, of the nature of the `ResultValidator`.